### PR TITLE
Added generic proper error handling

### PR DESCRIPTION
**Pull request by tboerger**

### CODE REVIEW

File: lib/crowbar/client/app/base.rb [hunk 1]
```diff
@@ -43,7 +43,20 @@ def say(message)
...
+              case
+              when options["format"] == "json" || options["json"]
+                JSON.pretty_generate(
+                  error: message
+                )
+              else
+                message
+              end
+            else
+              message
```
Comment (jdsn) [inline lib/crowbar/client/app/base.rb]: is it possible to combine these two else clauses?

Comment (tboerger) [inline lib/crowbar/client/app/base.rb]: I think it's better readable now :)

Comment (jdsn) [inline lib/crowbar/client/app/base.rb]: Just an idea to replace this block with:
```
def err(message, exit_code = nil)
  do_json = options["format"] && ( options["format"] == "json" || options["json"] )
  if do_json
    JSON.pretty_generate(error: message)
  else
    message
  end
end
```

**Comment by MaximilianMeister** [comment]
:+1: tested

**Comment by rsalevsky** [comment]
:+1:

**Comment by jdsn** [comment]
:-1: due to code duplication

**Comment by tboerger** [comment]
@jdsn I have resolved the code duplication now.

**Comment by jdsn** [comment]
still looks like code duplication, but I don't either know how to do it better :)
but this looks a lot better, so  :+1:

**Pull request merged**